app/models/ContributionAmount.scala

```diff
@@ -20,4 +19,6 @@ object ContributionAmount {
       currency <- Currency.fromString(currencyString)
     } yield ContributionAmount(number, currency)
   }
+
+  implicit val showContributionAmount: Show[ContributionAmount] = Show.show(f => f"${f.amount}%1.2f${f.currency}")
```

Comment:
Luckily this is a string that isn't displayed, or isn't planned to be displayed to the user. It is stored on the app, then sent to our campaign management tool to avoid showing membership campaign to user who paid.

So I went for the technical formatting:
50.0GBP or 25.0EUR